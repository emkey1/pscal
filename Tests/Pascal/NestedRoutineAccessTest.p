program NestedRoutineAccessTest;

var g: integer;

procedure Outer;
    var currentline: integer;
        outtext: string[20];

    procedure InnerProc;
    begin
        g := g + 1;
        currentline := currentline + 1;
        outtext := 'changed';
    end;

    function InnerFunc: integer;
    begin
        InnerProc;
        InnerFunc := g + currentline;
    end;

begin
    g := 5;
    currentline := 0;
    outtext := 'start';
    InnerProc;
    if InnerFunc = g + currentline then
        writeln('PASS: Nested routines can access outer vars')
    else
        writeln('FAIL: Nested routines cannot access outer vars');
    writeln('currentline=', currentline);
    writeln('outtext=', outtext);
end;

begin
    g := 10;
    Outer;
end.
