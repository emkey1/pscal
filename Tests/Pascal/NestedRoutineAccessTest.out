PASS: Nested routines can access outer vars
currentline=2
outtext=changed
